### PR TITLE
Refactor FXIOS-7301 - Remove 1 closure_body_length violation from ReaderPanel.swift

### PR DESCRIPTION
**Pull request by ionixjunior**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7301)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/16442)

## :bulb: Description
This PR removes 1 `closure_body_length` violation from the `ReaderPanel.swift` file by extracting the logic that creates the empty state view. I created a new group called `Reader` and moved `ReaderPanel.swift` into it. Then, I introduced a new file, `ReaderPanelEmptyStateView.swift`, where I implemented the empty state logic. Finally, I replaced the original empty state code in `ReaderPanel.swift` with a reference to the new file, resolving the `closure_body_length` violation.

Below follow the image reference of the empty state, before and after the implementation.

| Before | After | 
|---|---|
|  empty reading list - before  | empty reading list - after | 

This PR is part of a series of PRs described here https://github.com/mozilla-mobile/firefox-ios/issues/16442#issuecomment-2197676804.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/Client/Frontend/Library/Reader/ReaderPanel.swift [hunk 1]
```diff
@@ -26,7 +26,7 @@ private struct ReadingListTableViewCellUX {
     static let MarkAsReadButtonTitleEdgeInsets = UIEdgeInsets(top: 4, left: 4, bottom: 4, right: 4)
 }
 
-private struct ReadingListPanelUX {
+struct ReadingListPanelUX {
```
Comment (cyndichin) [inline firefox-ios/Client/Frontend/Library/Reader/ReaderPanel.swift:29]: thanks for cleaning this up! What do you think about moving this struct to the EmptyStateView and keeping it private so its self contained in that file?

File: firefox-ios/Client/Frontend/Library/Reader/ReaderPanelEmptyStateView.swift [hunk 2]
```diff
@@ -0,0 +1,167 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import UIKit
+import Common
+
+class ReaderPanelEmptyStateView: UIView {
```
Comment (cyndichin) [inline firefox-ios/Client/Frontend/Library/Reader/ReaderPanelEmptyStateView.swift:8]:
```suggestion
final class ReaderPanelEmptyStateView: UIView {
```

File: firefox-ios/Client/Frontend/Library/Reader/ReaderPanelEmptyStateView.swift [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    let windowUUID: WindowUUID
+    let themeManager: Common.ThemeManager
+
+    private lazy var welcomeLabel: UILabel = {
```
Comment (cyndichin) [inline firefox-ios/Client/Frontend/Library/Reader/ReaderPanelEmptyStateView.swift:12]: nit: to keep consistent with how we initialize other UI elements
```
private lazy var welcomeLabel: UILabel = .build {
```

**Review by cyndichin — COMMENTED** [review]
thanks @ionixjunior for the changes. left a couple of comments to consider, let me know what you think!

File: firefox-ios/Client/Frontend/Library/Reader/ReaderPanel.swift (hunk 1 above)
Comment (ionixjunior) [reply firefox-ios/Client/Frontend/Library/Reader/ReaderPanel.swift:29]: Sure, I completely agree. That makes sense and will keep the code more self-contained, reducing the risk of misuse. I'll make the changes.

File: firefox-ios/Client/Frontend/Library/Reader/ReaderPanelEmptyStateView.swift (hunk 3 above)
Comment (ionixjunior) [reply firefox-ios/Client/Frontend/Library/Reader/ReaderPanelEmptyStateView.swift:12]: Sure! Thanks for pointing that out, I'll make the change.

**Comment by ionixjunior** [comment]
Thank you for your suggestions, @cyndichin. I've applied all of them and rebased the branch.

File: firefox-ios/Client/Frontend/Library/Reader/ReaderPanelEmptyStateView.swift [hunk 4]
```diff
@@ -0,0 +1,166 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import UIKit
+import Common
+
+private struct ReadingListPanelUX {
```
Comment (cyndichin) [inline firefox-ios/Client/Frontend/Library/Reader/ReaderPanelEmptyStateView.swift:8]: nit: what do you think about adding this within the `UIView` class and follow the same convention in naming:
`private struct UX {` similar to other areas of the application?

**Review by cyndichin — APPROVED** [review]
@ionixjunior looks great! just one nit for consideration, let me know what you think!

File: firefox-ios/Client/Frontend/Library/Reader/ReaderPanelEmptyStateView.swift (hunk 4 above)
Comment (ionixjunior) [reply firefox-ios/Client/Frontend/Library/Reader/ReaderPanelEmptyStateView.swift:8]: Great suggestion! Thank you for pointing that out. Please continue to flag these things for me!

**Comment by ionixjunior** [comment]
@cyndichin thank you so much for your patience. I'm learning a lot from this project, and I'll work to improve my contributions in future PRs.

**Comment by FilippoZazzeroni** [comment]
Hi @ionixjunior i see tests are failing, please rebase on main, and then we will run it again. Thank you very much

**Comment by FilippoZazzeroni** [comment]
Thanks @ionixjunior by re running tests, it worked 🥳

**Comment by cyndichin** [comment]
> @cyndichin thank you so much for your patience. I'm learning a lot from this project, and I'll work to improve my contributions in future PRs.

thank you for your work as well @ionixjunior! It greatly helps us when you contribute and much appreciate :) 🔥 

Thanks @FilippoZazzeroni for merging the PR!

**Pull request merged**